src/quick-lint-js/fe/parse.cpp

```diff
@@ -402,6 +402,80 @@ void parser::error_on_invalid_as_const(expression* ast,
   }
 }
 
+void parser::error_on_pointless_compare_against_literal(
+    expression::binary_operator* ast) {
+  auto is_comparison_operator = [](string8_view s) -> bool {
+    return s == u8"=="sv || s == u8"==="sv || s == u8"!="sv || s == u8"!=="sv;
+  };
+  auto is_strict_operator = [](string8_view op) -> bool {
+    return op == u8"==="sv || op == u8"!=="sv;
+  };
+  auto is_regexp_literal = [](expression::literal* reg_exp) -> bool {
+    return !(reg_exp->span().begin() == reg_exp->span().end() ||
+             *reg_exp->span().begin() != u8'/');
```

Comment:
Nit: Avoid calling `span` many times:
```suggestion
    source_code_span span = reg_exp->span();
    return !(span.begin() == span.end() ||
             *span.begin() != u8'/');
```